pkg/operator/controller/externaldns/rbac_test.go

```diff
@@ -0,0 +1,625 @@
+				},
+				RoleRef: rbacv1.RoleRef{
+					APIGroup: "rbac.authorization.k8s.io",
+					Kind:     "ClusterRole",
+					Name:     controller.ExternalDNSBaseName,
+				},
+				Subjects: []rbacv1.Subject{
+					{
+						Kind:      "ServiceAccount",
+						Name:      "othersa",
+						Namespace: testNamespace,
+					},
+				},
+			},
+			inputExpected: &rbacv1.ClusterRoleBinding{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: controller.ExternalDNSResourceName(testExternalDNS),
+				},
+				RoleRef: rbacv1.RoleRef{
+					APIGroup: "rbac.authorization.k8s.io",
+					Kind:     "ClusterRole",
+					Name:     controller.ExternalDNSBaseName,
+				},
+				Subjects: []rbacv1.Subject{
+					{
+						Kind:      "ServiceAccount",
+						Name:      controller.ExternalDNSResourceName(testExternalDNS),
+						Namespace: testNamespace,
+					},
+				},
+			},
+			expectChanged: true,
+		},
+		{
+			name: "Subject namespace changed",
+			inputCurrent: &rbacv1.ClusterRoleBinding{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: controller.ExternalDNSResourceName(testExternalDNS),
+				},
+				RoleRef: rbacv1.RoleRef{
+					APIGroup: "rbac.authorization.k8s.io",
+					Kind:     "ClusterRole",
+					Name:     controller.ExternalDNSBaseName,
+				},
+				Subjects: []rbacv1.Subject{
+					{
+						Kind:      "ServiceAccount",
+						Name:      "othersa",
+						Namespace: testNamespace,
+					},
+				},
+			},
+			inputExpected: &rbacv1.ClusterRoleBinding{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: controller.ExternalDNSResourceName(testExternalDNS),
+				},
+				RoleRef: rbacv1.RoleRef{
+					APIGroup: "rbac.authorization.k8s.io",
+					Kind:     "ClusterRole",
+					Name:     controller.ExternalDNSBaseName,
+				},
+				Subjects: []rbacv1.Subject{
+					{
+						Kind:      "ServiceAccount",
+						Name:      controller.ExternalDNSResourceName(testExternalDNS),
+						Namespace: "otherns",
+					},
+				},
+			},
+			expectChanged: true,
+		},
+		{
+			name: "All fields changed",
+			inputCurrent: &rbacv1.ClusterRoleBinding{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: controller.ExternalDNSResourceName(testExternalDNS),
+				},
+				RoleRef: rbacv1.RoleRef{
+					APIGroup: "rbac.authorization.k8s.io",
+					Kind:     "ClusterRole",
+					Name:     "otherrole",
+				},
+				Subjects: []rbacv1.Subject{
+					{
+						Kind:      "ServiceAccount",
+						Name:      "othersa",
+						Namespace: "otherns",
+					},
+				},
+			},
+			inputExpected: &rbacv1.ClusterRoleBinding{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: controller.ExternalDNSResourceName(testExternalDNS),
+				},
+				RoleRef: rbacv1.RoleRef{
+					APIGroup: "rbac.authorization.k8s.io",
+					Kind:     "ClusterRole",
+					Name:     controller.ExternalDNSBaseName,
+				},
+				Subjects: []rbacv1.Subject{
+					{
+						Kind:      "ServiceAccount",
+						Name:      controller.ExternalDNSResourceName(testExternalDNS),
+						Namespace: namespace,
```

Comment:
I think you want `testNamespace`, which is defined in this file, not `namespace`, which is defined in `deployment_test.go`:
```suggestion
						Namespace: testNamespace,
```